### PR TITLE
Makes magboots effective in preventing gravitational movement effects, fixes magboots not working for singularity pull, eases the chainstun effect of overcharged supermatter

### LINKED ISSUE #41582: Singularity will pull you in regardless of being able to defy gravity (e.g. magboots)
This is caused by someone entirely not understanding how singularity_pull works.

### LINKED ISSUE #41611: CO2 delamming supermatter pulls ghosts
## Reproduction:
Start a CO2 delam. Hang arond it as a ghost, wait for the SM to pull you on top of it.

### PR DESCRIPTION
**Pull request by Barhandar**
Fixes #41582
Fixes #41611 (I hope)

:cl: Barhandar
fix: Magboots will yet again prevent singularity from pulling you in.
tweak: Magboots will now also prevent being pulled in by gravitational anomaly.
tweak: Magboots will also prevent effects of supermatter pull, gravitational catapult and RnD gravity gun.
balance: Being prone will now prevent knockdown-stun of supermatter pull, removing accumulating chainstunning that rendered CO2 delams impossible to escape by humans. You still can get chainstunned, but now you have a few moments between stuns to try and get out.
tweak: Said supermatter pull stun now has player feedback.
fix: Ghosts (and all other things that have no business being pulled) no longer get pulled by supermatter pull.
fix: RnD gravity gun and gravity catapult had their proton packs removed too and should no longer throw ghosts around.
/:cl:

I didn't make magboots turn you immune to singularity's pull back when it was simplegulo just to have some chumbis break it.

### CODE REVIEW

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -683,8 +683,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			var/atom/movable/pulled_object = P
 			if(ishuman(P))
 				var/mob/living/carbon/human/H = P
-				H.apply_effect(40, EFFECT_PARALYZE, 0)
-			if(pulled_object && !pulled_object.anchored && !ishuman(P))
+				if(H.incapacitated() || !(H.mobility_flags & MOBILITY_STAND))
+					return //You can't knock down someone who is already knocked down.
+				if(!H.mob_negates_gravity())
+					H.visible_message("<span class='danger'>[H] is suddenly knocked down, as if their legs have been pulled out from underneath them!</span>",\
```
Comment (ShizCalev) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
				H.visible_message("<span class='danger'>[H] is suddenly knocked down, as if [H.p_their()] [(H.get_num_legs() == 1) ? "leg has" : "legs have"] been pulled out from underneath [H.p_them()]!</span>",\
```

**Comment by Tlaltecuhtli** [comment]
make super singulo break your legs

**Comment by evsey9** [comment]
make it break your arms instead

**Comment by Iamgoofball** [comment]
:salt:

**Pull request merged**